lib/route.js

```diff
@@ -427,7 +432,8 @@ function buildRouting (options) {
       loggerOpts.serializers = context.logSerializers
     }
     const childLogger = createChildLogger(context, logger, req, id, loggerOpts)
-    childLogger[kDisableRequestLogging] = disableRequestLogging
+    const resolvedDisableRequestLogging = disableRequestLoggingFn ? disableRequestLoggingFn(context, req) : disableRequestLogging
```

Comment:
I'm not 100% convinced that we should be passing the context object. I believe that calling the function (if passed) with the constructed Fastify Request object should be enough to have access the APIs from Fastify (they can access the server instance from there as well).

Reply:
Yes, `context` should never be passed to users directly.

Reply:
sorry for the delay, will do the change this week